packages/tfchain_client/test/farms_test.dart

```diff
@@ -0,0 +1,101 @@
+import 'dart:io';
+import 'dart:math';
+
+import 'package:test/test.dart';
+import 'package:tfchain_client/generated/dev/types/tfchain_support/types/farm.dart';
+import 'package:tfchain_client/tfchain_client.dart';
+
+void main() {
+  group("Query Farms Test", () {
+    late QueryClient queryClient;
+    setUp(() async {
+      queryClient = QueryClient("wss://tfchain.dev.grid.tf/ws");
+      await queryClient.connect();
+    });
+
+    tearDownAll(() async {
+      await queryClient.disconnect();
+    });
+
+    test('Test Get Farm by Id', () async {
+      Farm? farm = await queryClient.farms.get(id: 1);
+      expect(farm, isNotNull);
+    });
+
+    test('Test Get Farm by wrong Id', () async {
+      try {
+        Farm? farm = await queryClient.farms.get(id: -50);
+      } catch (e) {
+        expect(e, isNotNull);
+      }
+    });
+  });
+
+  group("Test Farms", () {
+    late Client client;
+    final mnemonic = Platform.environment['MNEMONIC']!;
+    final String link =
+        Platform.environment['LINK'] ?? 'wss://tfchain.dev.grid.tf/ws';
+    final String type = Platform.environment['SIGN_TYPE'] ?? 'sr25519';
+
+    setUp(() async {
+      client = Client(link, mnemonic, type);
+      await client.connect();
+    });
+
+    tearDownAll(() async {
+      try {
+        await client.farms.removeFarmIp(farmId: 4588, ip: "198.165.15.25/16");
+      } catch (error) {
+        print("Error removing Ip from farm with id 4588: $error");
+      }
+      await client.disconnect();
+    });
+
+    test('Test create farm', () async {
+      final random = Random();
+      final farmName = 'farm_${random.nextInt(999)}';
```

Comment:
it can be more random than that and ensure there is no farm with this name before creating a new farm